src/main/java/com/klodnicki/Bike/v2/model/RentRequest.java

```diff
@@ -0,0 +1,14 @@
+package com.klodnicki.Bike.v2.model;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class RentRequest {
+
+    private Long userId;
+    private Long bikeId;
+    private Long chargingStationId;
```

Comment:
I think that this field is not actually required when you send request to the application. In the service layer you have logic that checks if we give correct id of a station, while Bike object already holds the value of station it is assigned to :) You can take the station from there and doesn't have to manually specify it in the RentRequest when sending such request

Reply:
yes and no at the same time. If I do like you said (If I understood correctly) then I must take id of a station from the bike like so:

![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/5d06f00b-cef1-4923-a06d-0ea9a33e67cb)

but in some circumstances (-> when bike is not set to the station ) I will receive nullPointerException
It is especially important during testing which I'm doing right now. You asked me to delete method for adding bike to the station list of bikes and currently its problematic to set the bike to the station just with JSON request.
So that is why that station field was created in RentRequest class.

Reply:
If bike is not set to the station it can't be rented. You just have to handle that situation and I stand by the principle that this field is not required.

Reply:
OK! done! I have created method to add bike to list at station, plus remade this field as you required.